File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int base = 0, size = t.length();
        if(s.length() < 1) return true;
        for (int i = 0; i < size && base < s.length(); i++){
            if (s.charAt(base) == t.charAt(i))
                base++;
        }
        return (base == s.length()) ? true : false;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int base = 0, size = t.size();
        for (int i = 0; i < size; i++)
        {
            if (s[base] == t[i]) {
                base++;
            }
        }
        return (base == s.size()) ? true : false;
    }
};